config/image-cache.js

```diff
@@ -47,35 +47,29 @@ async function getGcsAuthToken() {
       {
         method: 'GET',
         headers: { 'Metadata-Flavor': 'Google' },
+        throwOnError: true,
       }
     );
     const json = await response.body.json();
-    if (response.statusCode === 200 && json.access_token) {
-      // Cache the token so we can reuse it for other images.
-      process.env['IMAGE_CACHE_GCS_AUTH_TOKEN'] = json.access_token;
-      return json.access_token;
-    }
-    throw new Error(`GCS access token not found ${response.statusCode}: ${json}`);
+    // Cache the token so we can reuse it for other images.
+    process.env['IMAGE_CACHE_GCS_AUTH_TOKEN'] = json.access_token;
+    return json.access_token;
   } catch (error) {
-    throw new Error(`Error fetching GCS access token: ${error.message}`);
+    throw new Error(`GCS access token error: ${error}`);
```

Comment:
The `@google-cloud/storage` library can be used to handle the auth and upload a lot easier, it would be like 3-4 lines, I'm using it over here https://github.com/hirosystems/stacks-node-crawler/blob/720d4f9f79155a84a19c3d1028f0df1a54b3fbfb/src/index.ts#L5

E.g.
```ts
import { Storage } from '@google-cloud/storage';
const uploadStream = new Storage().bucket(myBucket).file(myfile).createWriteStream();
stream.pipe(uploadStream);
```